rust/src/lib.rs

```diff
@@ -8,6 +8,18 @@ mod bindings;
 #[repr(transparent)]
 pub struct Plugin(isize);
 
+impl From<isize> for Plugin {
+    fn from(i: isize) -> Self {
+        Plugin(i)
+    }
+}
+
+impl From<Plugin> for isize {
+    fn from(p: Plugin) -> Self {
+        p.0
```

Comment:
This doesn't actually work because `drop` is implemented for `Plugin` so it won't live beyond this function and the ID returned now pointed to a deallocated plugin.

The plugin lifetime probably needs to be managed manually using `std::mem::forget` and `drop` - or we could add a function to prevent `Plugin` from being freed on `drop`